Script improvements

tckb opened the pull request:
I've cleaned up the python script and added a helper script to download the release directly and flash into the printer. 

NB: I tested this on MacOs and should work on *nixes.

moonglow commented:
I love your idea to make firmware path as script command line option ( i'm not a python programmer hehe so i make it dumb as possible ), but it not works well for me i use Linux ( ubuntu ) and after printer was found script tries to send request but failed with timeout and printer continues to boot ( my script stops printer booting )... strange i will try to found a reason

tckb commented on ff_flash_firmware.py:
```diff
@@ -0,0 +1,103 @@
+printer.set_configuration()
+
+# add a bit delay to reduce timeouts
+sleep(5)
```
extra sleep to fix the timeout?

tckb replied:
@moonglow you can reduce this time to 1 sec to see if you can flash it in 1 go.

tckb commented:
@moonglow haha, I'm a seasonal developer but, not a firmware guy :P 

yeah I noticed it too, there's an initial timeout on the first try. The script takes at least two times.  I added an extra 'sleep' to see if that helped. Because it was too long, the printer continues to boot skipping this. I will try some more adjustments to see if it helps.

tckb commented:
@moonglow  I have added another fix, this seems to work directly.  Can you test it?

moonglow commented:
thank you, i will test it today

moonglow commented:
Very strange thing, first of all bootloader stops and this is good i can rerun your script many times, but without success :( I tries about  times, but if i start my old script ( without printer was rebooted ) it write firmware without any problem, my trace log:
```
Initiating control loop...
Starting control loop...
Traceback (most recent call last):
  File "./ff_flash_firmware.py", line 73, in <module>
    printer.set_configuration()
  File "/usr/local/lib/python3.7/dist-packages/usb/core.py", line 905, in set_configuration
    self._ctx.managed_set_configuration(self, configuration)
  File "/usr/local/lib/python3.7/dist-packages/usb/core.py", line 113, in wrapper
    return f(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/usb/core.py", line 159, in managed_set_configuration
    self.backend.set_configuration(self.handle, cfg.bConfigurationValue)
  File "/usr/local/lib/python3.7/dist-packages/usb/backend/libusb1.py", line 812, in set_configuration
    _check(self.lib.libusb_set_configuration(dev_handle.handle, config_value))
  File "/usr/local/lib/python3.7/dist-packages/usb/backend/libusb1.py", line 604, in _check
    raise USBError(_strerror(ret), ret, _libusb_errno[ret])
usb.core.USBError: [Errno 16] Resource busy
```

tckb commented:
@moonglow  my suspicion is how USB devices are handled in linux and macOs are different. I've added the extra `set_configuration` to  "emulate the re-running of the script. May be this is what tripping. I removed this extra config step, and this works just fine on macOs. 

can you retry this from scratch?

moonglow commented:
@tckb now it works for me from first time! i will do some checks for Windows and will merge it. thanks !

tckb commented:
@moonglow  we might need to add a tandem powershell or batch script for windows. I don't have a windows to test it out.  But i've heard with newer windows versions, you can run bash scripts :-/
